buildenv/jenkins/common/build.groovy

```diff
@@ -845,7 +845,8 @@ def build_all() {
                 timeout(time: 5, unit: 'HOURS') {
                     if ("${DOCKER_IMAGE}") {
                         // TODO: remove this workaround when https://github.com/adoptium/infrastructure/issues/3597 resolved. related: infra 9292
-                        if ((SDK_VERSION.toInteger() >= 17 && PLATFORM ==~ /x86-64_linux.*/) || (PLATFORM ==~ /ppc64le_linux.*/ )) {
+                        if ((PLATFORM ==~ /ppc64le_linux.*/) ||
+                            (PLATFORM ==~ /x86-64_linux.*/ && (SDK_VERSION.toLowerCase() == 'next' || SDK_VERSION.toInteger() >= 17))) {
```

Comment:
Parentheses are still missing, they should be around each binary expression, so it should be:
```
                            ((PLATFORM ==~ /x86-64_linux.*/) && ((SDK_VERSION.toLowerCase() == 'next') || (SDK_VERSION.toInteger() >= 17)))) {
```
I still don't think the use of `toLowerCase()` is necessary; if someone specified a bad `SDK_VERSION` it can still break (e.g. `SDK_VERSION = 'future'`).

Reply:
The parentheses are not simply "cosmetic", they help people reading the code in that they don't need to remember relative operator priorities (e.g. `==` vs. `||` or `>=`). See https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md which is primarily concerned with C/C++, but not all languages have the same operator precedence rules, making it more important to eliminate any confusion.

My example using `'future'` illustrates that we'll still have the possibility of `NumberFormatException` whether we use `toLowerCase()` or not.

Reply:
I will add parentheses then. 
But yes, `toLowerCase()` is not to address `NumberFormatException`. It a caveat for `non-lowercase` shapes of `next`.